### PR TITLE
Add support for VK_EXT_buffer_device_address and VK_EXT_memory_priority

### PR DESCRIPTION
**Pull request by jeffbolznv**
This is ready for review, I just need to split the commits.

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -36548,6 +36548,134 @@ TEST_F(VkLayerTest, AndroidHardwareBufferExporttBuffer) {
...
+    vkGetPhysicalDeviceFeatures2KHR(gpu(), &features2);
+    buffer_device_address_features.bufferDeviceAddressCaptureReplay = VK_FALSE;
+
+    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &features2));
+    ASSERT_NO_FATAL_FAILURE(InitRenderTarget());
+
+    PFN_vkGetBufferDeviceAddressEXT vkGetBufferDeviceAddressEXT =
+        (PFN_vkGetBufferDeviceAddressEXT)vkGetInstanceProcAddr(instance(), "vkGetBufferDeviceAddressEXT");
+
+    VkBufferCreateInfo buffer_create_info = {VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO};
```
[the remark this answers is not in the document]
Comment (jeffbolznv) [reply tests/layer_validation_tests.cpp]: fixed

File: layers/parameter_validation_utils.cpp [hunk 2]
```diff
@@ -2993,3 +2993,18 @@ void StatelessValidation::PostCallRecordDestroyRenderPass(VkDevice device, VkRen
...
+
+bool StatelessValidation::manual_PreCallValidateAllocateMemory(VkDevice device, const VkMemoryAllocateInfo *pAllocateInfo,
+                                                               const VkAllocationCallbacks *pAllocator, VkDeviceMemory *pMemory) {
+    bool skip = false;
+
+    if (pAllocateInfo) {
+        auto chained_prio_struct = lvl_find_in_chain<VkMemoryPriorityAllocateInfoEXT>(pAllocateInfo->pNext);
+        if (chained_prio_struct && (chained_prio_struct->priority < 0.0f || chained_prio_struct->priority > 1.0f)) {
+            skip |=
+                log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT, 0,
```
[the remark this answers is not in the document]
Comment (jeffbolznv) [reply layers/parameter_validation_utils.cpp]: fixed

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -15693,6 +15703,42 @@ VKAPI_ATTR void VKAPI_CALL DestroySamplerYcbcrConversionKHR(VkDevice device, VkS
...
+                                         "VUID-VkBufferDeviceAddressInfoEXT-buffer-02601", "vkGetBufferDeviceAddressEXT()",
+                                         "VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT_EXT");
+    }
+
+    return skip;
+}
+
+VKAPI_ATTR VkDeviceAddress VKAPI_CALL GetBufferDeviceAddressEXT(VkDevice device, const VkBufferDeviceAddressInfoEXT *pInfo) {
+    layer_data *dev_data = GetLayerDataPtr(get_dispatch_key(device), layer_data_map);
+    bool skip = PreCallValidateGetBufferDeviceAddressEXT(dev_data, pInfo);
```
[the remark this answers is not in the document]
Comment (jeffbolznv) [reply layers/core_validation.cpp]: I think I got this right, please double-check (ha, a locking joke!)

**Comment by jeffbolznv** [comment]
I believe I've addressed all the comments, and I rebased past the scalar_block_layout change.

File: layers/core_validation.cpp [hunk 4]
```diff
@@ -15698,6 +15708,48 @@ VKAPI_ATTR void VKAPI_CALL DestroySamplerYcbcrConversionKHR(VkDevice device, VkS
...
+
+VKAPI_ATTR VkDeviceAddress VKAPI_CALL GetBufferDeviceAddressEXT(VkDevice device, const VkBufferDeviceAddressInfoEXT *pInfo) {
+    layer_data *dev_data = GetLayerDataPtr(get_dispatch_key(device), layer_data_map);
+    unique_lock_t lock(global_lock);
+    bool skip = PreCallValidateGetBufferDeviceAddressEXT(dev_data, pInfo);
+    if (!skip) {
+        lock.unlock();
+        return dev_data->dispatch_table.GetBufferDeviceAddressEXT(device, pInfo);
+    }
+    return 0;
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: return VK_ERROR_VALIDATION_FAILED_EXT

Comment (jeffbolznv) [reply layers/core_validation.cpp]: The return value is a VkDeviceAddress, not a VkResult. Maybe that wasn't the right decision in retrospect, but it is a command that isn't supposed to fail under valid usage.

Comment (jzulauf-lunarg) [reply layers/core_validation.cpp]: You were correct in the first place. What's very odd is the (correct) use of a hard coded zero, as opposed to a symbolic name like VK_NULL_BUFFER_DEVICE_ADDRESS (which doesn't exist).

> A value of zero is reserved as a "null" pointer and must not be returned as a valid buffer device address

**Comment by mikes-lunarg** [comment]
I de-conflicted with #549

**Pull request merged**